aws/data_source_aws_vpc_ids_test.go

```diff
@@ -0,0 +1,90 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsVpcIDs_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsVpcIDsConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					//datasource aws_vpc_ids will list all the vpcs in a region, so it also includes the Default vpc.
+					//https://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/default-vpc.html
+					resource.TestCheckResourceAttr("data.aws_vpc_ids.all", "ids.#", "3"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsVpcIDs_tags(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsVpcIDsConfig_tags(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_vpc_ids.selected", "ids.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsVpcIDsConfig() string {
+	return fmt.Sprintf(`
+	provider "aws" {
```

Comment:
Declaring the "default" provider region is extraneous in test configurations and discouraged unless its absolutely necessary (e.g. EC2 Classic testing) -- its handled by the `AWS_DEFAULT_REGION` environment variable (or defaults to `us-west-2`) here: 

https://github.com/terraform-providers/terraform-provider-aws/blob/59ea1c3e859c45a3a19d60b49b7c42fee88e5d7b/aws/provider_test.go#L69